xbmc/video/VideoInfoScanner.cpp

```diff
@@ -1486,9 +1486,18 @@ namespace VIDEO
                                      movieDetails.m_iSeason, movieDetails.m_iEpisode, strTitle);
     }
 
-    if (!movieDetails.HasStreamDetails() &&
-        CServiceBroker::GetSettingsComponent()->GetSettings()->GetBool(
-            CSettings::SETTING_MYVIDEOS_EXTRACTFLAGS))
+    /* As HasStreamDetails() returns true for TV shows (because the scraper calls SetVideoInfoTag()
+     * directly to set the duration) a better test is just to see if we have any common flag info
+     * missing.  If we have already read an nfo file then this data should be populated, otherwise
+     * get it from the video file */
+
+    CStreamDetails strmdetails{movieDetails.m_streamDetails};
+
+    bool probeFile = strmdetails.GetVideoCodec(1).empty() || strmdetails.GetVideoHeight(1) == 0 ||
+                     strmdetails.GetVideoWidth(1) == 0 || strmdetails.GetVideoDuration(1) == 0;
+
+    if (probeFile && CServiceBroker::GetSettingsComponent()->GetSettings()->GetBool(
+                         CSettings::SETTING_MYVIDEOS_EXTRACTFLAGS))
     {
```

Comment:
You never reach this block if the setting value is false. So there is no need to construct and check the `CStreamDetails` instance if the setting value is false.

I suggest to optimize this code a bit:

```c++
if (CServiceBroker::GetSettingsComponent()->GetSettings()->GetBool(CSettings::SETTING_MYVIDEOS_EXTRACTFLAGS))
{
    const CStreamDetails strmdetails{movieDetails.m_streamDetails};
    if (strmdetails.GetVideoCodec(1).empty() || strmdetails.GetVideoHeight(1) == 0 || strmdetails.GetVideoWidth(1) == 0 || strmdetails.GetVideoDuration(1) == 0)
    {
        ...
```

Reply:
Yes.  The TV scraper creates a single videostream with just the duration set in it if/when it has scraped a value for an episode and it does this exactly once for each episode in turn.  If there is no duration and therefore no videostream then all of the calls will return either zero or an empty string and the data will be obtained from the file.

Thanks for the code optimisation :)